zasilkovna.php

```diff
@@ -1208,4 +1208,38 @@ public function getTailBlockHtml(\VirtueMartModelZasilkovna\CheckoutModules\Abst
         return $this->renderer->renderToString();
     }
 
+    /**
+     * @param TableOrders &$order
+     * @param string $old_order_status
+     * @return void|null
+     * @throws Exception
+     */
+    public function plgVmOnUpdateOrderPayment(&$order, $old_order_status)
```

Comment:
musí tam být parametr `$old_order_status`? musí být u `$order` "&"?

Reply:
jako nemusi, ale stejne je to asi v 8mi jinych pluginech primo od Vm, tak bych to tak nechal

Reply:
tak to tedy udělejme podle volání v administrator/components/com_virtuemart/models/orders.php - &$data, $old_order_status, $inputOrder

metoda by měla vracet jen void, aby to šlo případně typovat (sice VM počítá s array, ale nemusíme teď řešit)

Reply:
Promin, jak jsi prisel na to, ze metoda by mela vracet void?
mrkni treba vmpayment/amazon.php, vmpayment/paypal.php, vmpayment/paypal_checkout.php a dalsi

vraci null, bool, void. ja mam v anotaci jen to, co skutecne vraci nase metoda

Reply:
Není důvod, aby vracela dva různé typy. A kdybych si měl momentálně vybrat, volím void, např. protože ten "vrací" když doběhne až nakonec. Že ostatní také vrací různé (náhodné?) typy se mi nezdá jako dobrý důvod, proč to dělat také.

Reply:
ti "ostatni" jsou tvurci VM. ale at je po tvym.